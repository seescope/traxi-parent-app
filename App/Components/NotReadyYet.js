import React from 'react';
import { Image, Dimensions, Text, Platform, View, TextInput } from 'react-native';
import { WHITE, NEUTRAL, TRANSPARENT, LIGHTEST_GREY } from '../Constants/Colours';
import Spacing from '../Components/Spacing';
import Button from '../Components/Button';
import { isIOS } from '../Utils';

const { width } = Dimensions.get('window');

const containerStyle = {
  backgroundColor: NEUTRAL,
  flex: 1,
  paddingHorizontal: 32,
  paddingTop: 32,
};

const headerStyle = {
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: Platform.select({
    ios: 26,
    android: 24,
  }),
  textAlign: 'center',
  fontFamily: 'Raleway-ExtraBold',
};

const bodyStyle = {
  paddingBottom: 16,
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: Platform.select({
    ios: 16,
    android: 14,
  }),
  textAlign: 'left',
  fontFamily: 'Raleway-Regular',
};

const androidInputStyle = {
  color: WHITE,
  height: 42,
  width: width - 64,
};

const iosInputStyle = {
  color: NEUTRAL,
  paddingHorizontal: 8,
  height: 32,
  width: width - 64,
  backgroundColor: LIGHTEST_GREY,
  borderRadius: 4,
};

const textInputStyle = Platform.select({
  ios: iosInputStyle,
  android: androidInputStyle,
});

const buttonContainer = {
  alignItems: 'center',
};

const topContainerStyle = {
  justifyContent: 'space-between',
  flexDirection: 'row',
};

export default () => (
  <View style={containerStyle}>
    <View style={topContainerStyle}>
      {isIOS ? <Image source={require('../Images/chevron_left.png')} /> : <View />}
      <Text style={headerStyle}>
        Not ready yet?
      </Text>
      <View />
    </View>

    <Spacing height={32} />

    <Text style={bodyStyle}>
      If your kid isn't around right now or you don't have time to set up traxi, don't worry.
    </Text>

    <Text style={bodyStyle}>
      Enter your phone number below and we'll remind you tomorrow.
    </Text>

    <Text style={bodyStyle}>
      You can also come back to the app whenever you're ready.
    </Text>

    <Spacing height={16} />

    <TextInput
      keyboardType="phone-pad"
      underlineColorAndroid={WHITE}
      style={textInputStyle}
    />

    <Spacing height={32} />

    <View style={buttonContainer}>
      <Button onPress={() => {}}>
        Remind me tomorrow
      </Button>
    </View>
  </View>
);
