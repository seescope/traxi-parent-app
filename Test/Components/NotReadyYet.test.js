import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import NotReadyYet, { remindMeTomorrow } from '../../App/Components/NotReadyYet';

jest.mock('../../App/Actions/SendPhoneNumberToSlack', () => () => Promise.resolve());

it('renders the <NotReadyYet> component', () => {
  const store = configureMockStore()();
  const tree = renderer.create(
    <Provider store={store}>
      <NotReadyYet />
    </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('posts a number to Slack', () => {
  const mockDispatch = jest.fn(() => Promise.resolve());
  remindMeTomorrow('0401633346')(mockDispatch);
  expect(mockDispatch.mock.calls).toMatchSnapshot();
});
