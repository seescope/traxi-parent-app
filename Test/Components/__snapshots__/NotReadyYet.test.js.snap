exports[`test renders the <NotReadyYet> component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(75, 73, 73, 1)",
      "flex": 1,
      "paddingHorizontal": 32,
      "paddingTop": 32,
    }
  }>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1} />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 26,
          "textAlign": "center",
        }
      }>
      [missing \"en.notReadyYet.header\" translation]
    </Text>
    <View />
  </View>
  <View
    style={
      Object {
        "height": 32,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 16,
        "paddingBottom": 16,
        "textAlign": "left",
      }
    }>
    [missing \"en.notReadyYet.body1\" translation]
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 16,
        "paddingBottom": 16,
        "textAlign": "left",
      }
    }>
    [missing \"en.notReadyYet.body2\" translation]
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 16,
        "paddingBottom": 16,
        "textAlign": "left",
      }
    }>
    [missing \"en.notReadyYet.body3\" translation]
  </Text>
  <View
    style={
      Object {
        "height": 16,
      }
    } />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }>
        [missing \"en.notReadyYet.button\" translation]
         
      </Text>
    </View>
  </View>
</View>
`;
